### ISSUE TITLE
Order owner_id different from User id

### ISSUE THREAD

**Issue by magosla** [mollie/laravel-cashier-mollie#14]
After a payment I noticed the list of orders returns null for the user.  The order table have a different value for the owner_id while the payments table has the actual owner_id. Below is an example

```
Orders table:
- id: 102, owner_id: 00c4439c-280b-4218-9f73-527563033462, owner_type: App\Models\User
- id: 101, owner_id: 009f6a86-a2bd-4ef9-b60f-0766cd71fc62, owner_type: App\Models\User

OrderItems table:
- order_id: 102, owner_id: 00c4439c-280b-4218-9f73-527563033462, owner_type: App\Models\User
- order_id: 101, owner_id: 009f6a86-a2bd-4ef9-b60f-0766cd71fc62, owner_type: App\Models\User

Payments table:
- order_id: 102, owner_id: ad10ef22-cb8a-413c-9072-085929ca6c0d, owner_type: App\Models\User
- order_id: 101, owner_id: 9c641eb3-47d3-4b9f-8335-cce386b209d3, owner_type: App\Models\User

Users table:
- id: ad10ef22-cb8a-413c-9072-085929ca6c0d
- id: 9c641eb3-47d3-4b9f-8335-cce386b209d3

**Comment by sandervanhooft** [comment]
Thanks for reporting @magosla .

May be an issue with using the uuids, we don't test for these, as they're not used by default. We'll look into it.

**Comment by sandervanhooft** [comment]
@magosla have you set a custom `primaryKey` on the user model by any chance?

**Comment by sandervanhooft** [comment]
@ciungulete I suspect this is caused by the new int cast on the owner id. Can you take a look?

**Comment by magosla** [comment]
> @ciungulete I suspect this is caused by the new int cast on the owner id. Can you take a look?

The owner_id field for the orders table has its type as CHAR 32

**Comment by ciungulete** [comment]
Solved here #16.

Also need to use  `protected $keyType = 'string';` in your billable model

**Comment by sandervanhooft** [comment]
@magosla Can you confirm the current main branch works for you?

**Comment by sandervanhooft** [comment]
Fixed in `v2.0.2` ([link](https://github.com/mollie/laravel-cashier-mollie/releases/tag/v2.0.2))

**Comment by magosla** [comment]
Yes it has been fixed, Thanks
